core/server/master/src/main/java/alluxio/master/file/InodeTtlChecker.java

```diff
@@ -60,83 +62,109 @@ public InodeTtlChecker(FileSystemMaster fileSystemMaster, InodeTree inodeTree) {
-                // public delete method will lock the path, and check WRITE permission required at
-                // parent of file
-                if (inode.isDirectory()) {
-                  mFileSystemMaster.delete(path,
-                          DeleteContext.mergeFrom(DeletePOptions.newBuilder()
-                                  .setRecursive(true).setAlluxioOnly(true)));
-                } else {
-                  mFileSystemMaster.delete(path,
-                          DeleteContext.mergeFrom(DeletePOptions.newBuilder()
-                                  .setAlluxioOnly(true)));
-                }
-                break;
-              default:
-                LOG.error("Unknown ttl action {}", ttlAction);
+                  processedFiles += 1;
+                  mTtlBuckets.remove(inode);
+                  break;
+                case DELETE:
+                  // public delete method will lock the path, and check WRITE permission required
+                  // at parent of file
+                  if (inode.isDirectory()) {
+                    mFileSystemMaster.delete(path,
+                        DeleteContext.mergeFrom(DeletePOptions.newBuilder().setRecursive(true)),
+                        journalContext);
+                    processedFiles += ((InodeDirectory) inode).getChildCount() + 1;
+                  } else {
+                    mFileSystemMaster.delete(path, DeleteContext.defaults(), journalContext);
+                    processedFiles += 1;
+                  }
+                  break;
+                case DELETE_ALLUXIO: // Default: DELETE_ALLUXIO
+                  // public delete method will lock the path, and check WRITE permission required at
+                  // parent of file
+                  if (inode.isDirectory()) {
+                    mFileSystemMaster.delete(path,
+                        DeleteContext.mergeFrom(DeletePOptions.newBuilder()
+                            .setRecursive(true).setAlluxioOnly(true)), journalContext);
+                    processedFiles += ((InodeDirectory) inode).getChildCount() + 1;
+                  } else {
+                    mFileSystemMaster.delete(path,
+                        DeleteContext.mergeFrom(DeletePOptions.newBuilder()
+                            .setAlluxioOnly(true)), journalContext);
+                    processedFiles += 1;
+                  }
+                  break;
+                default:
+                  LOG.error("Unknown ttl action {}", ttlAction);
+              }
+            } catch (Exception e) {
+              LOG.error("Exception trying to clean up {} for ttl check", inode, e);
             }
-          } catch (Exception e) {
-            LOG.error("Exception trying to clean up {} for ttl check", inode, e);
           }
         }
       }
+      success = true;
+    } catch (UnavailableException e) {
+      LOG.warn("Journal is not available. Cannot perform TTL checks.");
+    }
```

Comment:
flush leftover processedFiles?

Reply:
leftovers will be flushed when JournalContext is closed in the try-with-resource, i'm merely triggering it regularly to prevent too many being in the queue

Reply:
👌🏻 got it